kpatch-build/kpatch-build

```diff
@@ -1054,6 +1060,35 @@ if [[ -n "$CONFIG_DEBUG_INFO_BTF" ]]; then
 	fi
 fi
 
+if [[ -n "$CONFIG_LTO_CLANG" ]]; then
+	[[ -n "$CONFIG_LTO_CLANG_THIN" ]] || die "Only thin lto is supported. Try enable CONFIG_LTO_CLANG_THIN."
+
+	# With CONFIG_LTO_CLANG, vmlinux has LOCAL symbols for some GLOBAL
+	# functions. For example:
+	#
+	# readelf -s vmlinux  | grep perf_event_bpf_event
+	#
+	# non-LTO:
+	#  124946: ffffffff81287070   858 FUNC    GLOBAL DEFAULT    1 perf_event_bpf_event
+	#
+	# LTO:
+	#  122028: ffffffff81298cd0   861 FUNC    LOCAL  HIDDEN     1 perf_event_bpf_event
+	#
+	# To work around this, use vmlinux.o instead.
+	[[ -e "$VMLINUX" ]] || die "For kernel with CONFIG_LTO_CLANG, please supply vmlinux.o via -v|--vmlinux option."
+
+	# This is a heuristic: use -x to check vmlinux vs. vmlinux.o
+	[[ -x "$VMLINUX" ]] && die "For kernel with CONFIG_LTO_CLANG, please supply vmlinux.o instead of vmlinux via -v|--vmlinux option."
+
+	cp -f "$KERNEL_SRCDIR/Makefile" "$TEMPDIR/Makefile" || die
+	sed -i "s/--thinlto-cache-dir=\$(extmod_prefix).thinlto-cache/--lto-obj-path=vmlinux.o.thinlto.o/g" "$KERNEL_SRCDIR"/Makefile
+
+	cp -f "$KERNEL_SRCDIR/scripts/Makefile.build" "$TEMPDIR/Makefile.build" || die
+	sed -i "s/\$(ld_flags)/\$(ld_flags) --lto-obj-path=\$@.thinlto.o/g" "$KERNEL_SRCDIR"/scripts/Makefile.build
+
+	export KPATCH_CC_HANDLE_LTO=1
```

Comment:
I don't think "HANDLE" adds anything, can we just call it KPATCH_CC_LTO?

Reply:
Sure. Fixed.